transformer_engine/jax/cpp_extensions/attention.py

```diff
@@ -923,26 +923,30 @@ def check_supported(self):
         header = "Context parallel fused attention"
 
         allowed_layouts = [NVTE_QKV_Layout.NVTE_BSHD_BS2HD, NVTE_QKV_Layout.NVTE_BSHD_BSHD_BSHD]
-        assert self.config.qkv_layout in allowed_layouts, (
-            f"{header} only supports layouts: {','.join([str(x) for x in allowed_layouts])} got:"
-            f" {self.config.qkv_layout}"
-        )
+        if self.config.qkv_layout not in allowed_layouts:
```

Comment:
Hi, why do we replace `assert` with `if not` here?

Reply:
In a previous review it was suggested that asserts could be disabled and we should instead raise a proper exception.